src/correctness.cpp

```diff
@@ -0,0 +1,120 @@
+#include <array>
+
+#include <piwcsprwmodel/algorithms.h>
+
+#include "correctness.h"
+
+namespace piwcs::prw {
+
+bool isLocallyCorrect(const Model& model, IdRef id){
+	const Node* node = model.node(id);
+
+	std::array<bool, Node::MAX_SLOTS> isInward;
+	std::array<bool, Node::MAX_SLOTS> isOutward;
+	isInward.fill(false);
+	isOutward.fill(false);
+
+	auto sectionAt = [&](SlotId slot) -> const Section*{
+	    return model.section(node->section(slot));
```

Comment:
I think it would be better to check if `model.section` returns `nullptr` and return `false` as a node with empty slot signifies that `model` is incomplete. See 574bd5a39fc3273895762c5b3d6b2128937c9202.

Reply:
Hello, sorry for the long delay in working on this PR.
I have reverted 574bd5a39fc3273895762c5b3d6b2128937c9202 and implemented necessary changes in d3d4ade1d649bb73d87c8ea077680c4789cc50ac and 7cb61817f22cb23640dc7e049500b287ec24319f correspondingly. Now all section pointers are cached in `std::array` and on encountering `nullptr` among them function returns `false`. I also added runtime assertion to check if section id is `ID_INVALID` which should help to trace errors in case by-slot loop is incorrect.